### PR TITLE
Debugger Scaffolding and Command Line Arg changes

### LINKED ISSUE #25: [Debugger] Basic Scaffolding
Setup for Debugger

### PR DESCRIPTION
**Pull request by venkat24**
This PR adds a basic outline for the Debugger class, and also adds a command line flag for enabling the debugger.

Command line arg parsing has been revamped to use the [cxxopts](https://github.com/jarro2783/cxxopts) library.

A bunch of `CMakeLists.txt` files have been simplified with shorter directory includes. All modules now include headers from all others for convenience. 

Closes #25

### CODE REVIEW

File: src/cpu/src/opcodes.cpp [hunk 1]
```diff
@@ -376,14 +376,15 @@ void CPU::op_rlc(IReg *reg) {
 
 void CPU::op_rlc(Address addr) {
 	auto value = memory->read(addr);
-	auto msb = static_cast<bool>(value >> 7);
+	bool msb = value & (1 << 7);
```
Comment (trickarcher) [inline src/cpu/src/opcodes.cpp:379]: Can you explain what have you exactly done here?

File: src/debugger/src/debugger.cpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    : gameboy(std::move(gameboy)) {
+
+	// TODO
+	cpu = nullptr;
+	gpu = nullptr;
+	cartridge = nullptr;
+}
+
+void Debugger::tick() {
+	ticks++;
```
Comment (trickarcher) [inline src/debugger/src/debugger.cpp:21]: By this do you mean to increment the Debugger ticks? Is this in any way related to the CPU ticks?

File: src/main/main.cpp [hunk 3]
```diff
@@ -3,29 +3,71 @@
...
+		cout << cmdline_args_parser.help();
+		exit(0);
+	}
+
+	// Get ROM path and fail with help message if arg not parsed
+	string rom_path = "";
+	try {
+		rom_path = parsed_args["rom"].as<string>();
+	} catch (exception &e) {
+		cout << cmdline_args_parser.help();
```
Comment (trickarcher) [inline src/main/main.cpp:51]: Pretty Cool. This will make the entire thing so much more clean.

**Review by trickarcher — APPROVED** [review]
Hey @venkat24,
These commits are good to go. I had some questions regarding the debugger. Loved the idea of a command line argument system for accessing different options of the Emulator. The outline for the debugger is pretty good, although we would still have to decide on what kind of debugger model you would want. You mentioned something of a `socket` like interface. We can explore that later. As for this PR, everything's fine. We can merge it.

File: src/debugger/src/debugger.cpp (hunk 2 above)
Comment (venkat24) [reply src/debugger/src/debugger.cpp:21]: Hmm, I guess my language is a bit ambiguous here.

There's a distinction between ticks and cycles that I'm trying to make, where ticks is the number of times that `tick` is called, and CPU Cycles are a completely unrelated number, describing the number of real hardware cycles that should have been run at any particular time.

This is why I made the CpuCycles type, but I guess it's still a bit misleading. I'll try to improve this in coming PRs. Great point, thanks!

File: src/main/main.cpp (hunk 3 above)
Comment (venkat24) [reply src/main/main.cpp:51]: Yep, this library cxxopts is quite nice.

File: src/cpu/src/opcodes.cpp (hunk 1 above)
Comment (venkat24) [reply src/cpu/src/opcodes.cpp:379]: This opcode was setting the zero flag incorrectly, so I thought I'd make a quick fix for that. This line change is just to make it a tad more readable.

File: src/debugger/src/debugger.cpp (hunk 2 above)
Comment (trickarcher) [reply src/debugger/src/debugger.cpp:21]: Okay, I think I kind of get it now.

File: src/cpu/src/opcodes.cpp (hunk 1 above)
Comment (trickarcher) [reply src/cpu/src/opcodes.cpp:379]: Got it.

**Pull request merged**